src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorNodeWriter.cs

```diff
@@ -49,5 +53,169 @@ public sealed override void WriteCSharpCodeAttributeValue(CodeRenderingContext c
             context.Diagnostics.Add(BlazorDiagnosticFactory.Create_CodeBlockInAttribute(node.Source, content));
             return;
         }
+
+
+        // Currently the same for design time and runtime
+        public void WriteComponentTypeInferenceMethod(CodeRenderingContext context, ComponentTypeInferenceMethodIntermediateNode node)
```

Comment:
I like how this method is implemented with a combination of recent C# features yet also is structured for a really high level of clarity (e.g., repeating certain blocks instead of adding further abstractions in the name of DRY).

Reply:
https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction